### PR TITLE
C350420 Verify Browse Subjects field data validation (spitfire) (TaaS)

### PR DESCRIPTION
**Pull request by andreardztEPAM**
[C350420](https://foliotest.testrail.io/index.php?/cases/view/350420)

https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/5109/allure/

### CODE REVIEW

File: cypress/e2e/inventory/search/browse-search-subjects.cy.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  it(
+    'C350420 Verify Browse Subjects field data validation (spitfire) (TaaS)',
+    { tags: [testTypes.criticalPath, DevTeams.spitfire] },
+    () => {
+      InventorySearchAndFilter.switchToBrowseTab();
+      InventorySearchAndFilter.verifyKeywordsAsDefault();
+      BrowseSubjects.searchBrowseSubjects(testData.subjectName);
+      BrowseSubjects.checkRowValueIsBold(5, testData.subjectName);
+      InventorySearchAndFilter.clickResetAllButton();
```
Comment (zentestuken) [inline cypress/e2e/inventory/search/browse-search-subjects.cy.js:38]: There is no need to do a reset. A field just have to be cleared (input '' or click "x" in input field). There is a risk that reset would yield expected outcome while simply leaving the field blank not

File: cypress/e2e/inventory/search/browse-search-subjects.cy.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import testTypes from '../../../support/dictionary/testTypes';
+import DevTeams from '../../../support/dictionary/devTeams';
+import Users from '../../../support/fragments/users/users';
+import InventoryInstances from '../../../support/fragments/inventory/inventoryInstances';
+import InventorySearchAndFilter from '../../../support/fragments/inventory/inventorySearchAndFilter';
+import BrowseSubjects from '../../../support/fragments/inventory/search/browseSubjects';
+
+describe('Subject Browse', () => {
+  const testData = {
+    subjectName: 'Rock music--1961-1970',
```
Comment (zentestuken) [inline cypress/e2e/inventory/search/browse-search-subjects.cy.js:12]: There is no guarantee that a record with this subject value will be available on environment when testing. Could you please add steps to create a new instance with certain subject and then search for it, or get subjects via API to know what is available?

File: cypress/e2e/inventory/search/browse-search-subjects.cy.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    'C350420 Verify Browse Subjects field data validation (spitfire) (TaaS)',
+    { tags: [testTypes.criticalPath, DevTeams.spitfire] },
+    () => {
+      InventorySearchAndFilter.switchToBrowseTab();
+      InventorySearchAndFilter.verifyKeywordsAsDefault();
+      BrowseSubjects.searchBrowseSubjects(testData.subjectName);
+      BrowseSubjects.checkRowValueIsBold(5, testData.subjectName);
+      InventorySearchAndFilter.clickResetAllButton();
+      InventorySearchAndFilter.verifySearchButtonDisabled();
+      InventorySearchAndFilter.switchToSearchTab();
```
Comment (zentestuken) [inline cypress/e2e/inventory/search/browse-search-subjects.cy.js:40]: At this step we need to repeat action from Step 1 (browse with existing subject value)
Sorry, the wording in test case was not good. I have updated it now for clarity

**Review by zentestuken — CHANGES_REQUESTED** [review]

File: cypress/e2e/inventory/search/browse-search-subjects.cy.js (hunk 1 above)
Comment (andreardztEPAM) [reply cypress/e2e/inventory/search/browse-search-subjects.cy.js:38]: One of the validation from the test is that once we clear data after searching, the button is disabled. Not sure if it's better to remove the text or reset

File: cypress/e2e/inventory/search/browse-search-subjects.cy.js (hunk 2 above)
Comment (andreardztEPAM) [reply cypress/e2e/inventory/search/browse-search-subjects.cy.js]: Added api function to create the testdata needed

File: cypress/e2e/inventory/search/browse-search-subjects.cy.js (hunk 1 above)
Comment (andreardztEPAM) [reply cypress/e2e/inventory/search/browse-search-subjects.cy.js]: removed reset all and added clear search function

File: cypress/e2e/inventory/search/browse-search-subjects.cy.js (hunk 3 above)
Comment (andreardztEPAM) [reply cypress/e2e/inventory/search/browse-search-subjects.cy.js:54]: will resolve this after our conversation in teams

**Review by zentestuken — APPROVED** [review]

**Review by Scorpibear — APPROVED** [review]

**Pull request merged**